2021/CifarelliDiFranco/utils.py

```diff
@@ -0,0 +1,153 @@
+    parser.add_argument("-pa", "--postanalysis", action="store_true", help="applies an analysis to the .pcap file generated by the execution")
+    parser.add_argument("-ign", "--ignore", action="store_true", help="ignores dns traffic addressed to this machine (PLEASE NOTE: setting this flag will cause Tshark's capture to be faster)")
+    return parser.parse_args()
+
+def print_dns_analysis(qht, rht, ql, rl):
+    print(" DNS queries:")
+    for id in qht.keys():
+        print(id+" "+qht[id])
+    print("------------ total of "+str(len(qht.keys()))+" queries")
+    print("\n DNS reponses:")
+    for id in rht.keys():
+        print(id+" "+rht[id])
+    print("------------ total of "+str(len(rht.keys()))+" responses")
+    total = 0
+    print("\n DNS queries (retransmitted):")
+    for id in ql.keys():
+        print(id+": "+ql[id])
+        total += ql[id].count(" ")
+    print("------------ total of "+str(total)+" queries")
+    total = 0
+    print("\n DNS reponses (retransmitted):")
+    for id in rl.keys():
+        print(id+": "+rl[id])
+        total += rl[id].count(" ")
+    print("------------ total of "+str(total)+" responses")
+
+def check_dns_rcode(rcode, framenum):
+    global noerror
+    global formerr
+    global servfail
+    global nxdomain
+    global notimp
+    global refused
+    global yxdomain
+    global xrrset
+    global notauth
+    global notzone
+    global wrong_dns_frames
+
+    if rcode == NOERROR:
+        noerror += 1
+    elif rcode == FORMERR:
+        formerr += 1
+    elif rcode == SERVFAIL:
+        servfail += 1
+    elif rcode == NXDOMAIN:
+        nxdomain += 1
+    elif rcode == NOTIMP:
+        notimp += 1
+    elif rcode == REFUSED:
+        refused += 1
+    elif rcode == YXDOMAIN:
+        yxdomain += 1
+    elif rcode == XRRSET:
+        xrrset += 1
+    elif rcode == NOTAUTH:
+        notauth += 1
+    else:
+        notzone += 1
+    
+    wrong_dns_frames[rcode] += " "+str(framenum)
+
+def print_dns_rcodes():
+    print("\n Report of DNS codes found:")
+    print(str(noerror)+" no error (0) codes:"+wrong_dns_frames[str(0)])
+    print(str(formerr)+" format error code (1) codes:"+wrong_dns_frames[str(1)])
+    print(str(servfail)+" server fail (2) codes:"+wrong_dns_frames[str(2)])
+    print(str(nxdomain)+" no such name (3) codes:"+wrong_dns_frames[str(3)])
+    print(str(notimp)+" not implemented function (4) codes:"+wrong_dns_frames[str(4)])
+    print(str(refused)+" refused query (5) codes:"+wrong_dns_frames[str(5)])
+    print(str(yxdomain)+" yxdomain (6) codes (a name that shouldn't exists instead exists):"+wrong_dns_frames[str(6)])
+    print(str(xrrset)+" xrrset (7) codes (a rrset that shouldn't exists instead exists):"+wrong_dns_frames[str(7)])
+    print(str(notauth)+" notauth (8) codes:"+wrong_dns_frames[str(8)])
+    print(str(notzone)+" notzone (9) codes:"+wrong_dns_frames[str(9)])
+
+def get_net_info(interface):
```

Comment:
Fatta in questo modo e' assai complessa

Reply:
Cercheremo di cambiarlo, al massimo lo faremo passare come argomento da args